### PR TITLE
[supervisor] Remove old gp bin before symlink

### LINKED ISSUE #6533: deploy gp cli independent
In prod we still build images with mk2, which ships `gp` as part of the Gitpod layer. In preparation for mk3 we started to ship `gp` with supervisor:
https://github.com/gitpod-io/gitpod/blob/d3ce797ae7159a30b22b8c1436d3398435299c62/components/supervisor/pkg/supervisor/supervisor.go#L390-L399

If the gitpod-layer is present, `/usr/bin/gp` will be present which makes the symlink operation fail.

To ship the `gitpod-cli` independently of the gitpod-layer even with mk2, we could just delete the `to` path prior to symlinking, i.e.

```Go
	 for k, v := range binaries {
		var (
			from = filepath.Join(base, k)
			to   = filepath.Join("/usr/bin", v)
		)

		// New code
		_ = os.Remove(to)
		// ends here

		err = os.Symlink(from, to)
		if err != nil && !os.IsExist(err) {
			log.WithError(err).WithField("from", from).WithField("to", to).Warn("cannot create symlink")
		}
	}
```

Error checking is left as an exercise to the reader. Beware that `os.IsNotExist(err)` is a valid case if there's no gitpod-layer.

### PR DESCRIPTION
**Pull request by corneliusludmann**
## Description
This PR makes sure that the `gp` binary that is shipped with the supervisor is used even when a `gp` bin is already present in `/usr/bin`. This allows to ship a new `gp` version even with `image-builder` mk2.

## Related Issue(s)

Fixes #6533

## How to test

I tested this with `image-builder` as well as `image-builder-mk3` like this:
- Edit [server-config](https://console.cloud.google.com/kubernetes/configmap/europe-west1-b/core-dev/staging-clu-deploy-gp-cli-6533/server-config/details?project=gitpod-core-dev) and set `imageBuilderAddr` to `"image-builder:8080"` resp. `"image-builder-mk3:8080"`
- Restart server after changing the configmap: `$ kubectl rollout restart deployment server`
- Start a workspace
- Make sure there is no error/warning regarding creating the symlink, e.g. use this search query in the GCP Logs Explorer:
  ```
  resource.labels.namespace_name="staging-clu-deploy-gp-cli-6533"
  resource.labels.container_name="workspace"
  "symlink"
  ```
  as well as that `gp` works.

## Release Notes

```release-note
NONE
```

### CODE REVIEW

**Comment by roboquat** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by:
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please ask for approval from **corneliusludmann** after the PR has been reviewed.

Associated issue: *#6533*

The full list of commands accepted by this bot can be found [here](https://prow.gitpod-dev.com/command-help?repo=gitpod-io%2Fgitpod).

<details open>
Needs approval from an approver in each of these files:

- **[components/supervisor/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/supervisor/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Review by jeanp413 — APPROVED** [review]
Followed the "how to test" steps and there are no errors.

/lgtm

**Comment by roboquat** [comment]
LGTM label has been added.  <details>Git tree hash: cbd24b874dccd28b1c28c3aa1f4afbeb0db038fb</details>

**Comment by roboquat** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *<a href="https://github.com/gitpod-io/gitpod/pull/6648#pullrequestreview-805323051" title="LGTM">jeanp413</a>*

Associated issue: *#6533*

The full list of commands accepted by this bot can be found [here](https://prow.gitpod-dev.com/command-help?repo=gitpod-io%2Fgitpod).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[components/supervisor/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/supervisor/OWNERS)~~ [jeanp413]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Pull request merged**
